static/app/utils/replays/hooks/useReplayLayout.tsx

```diff
@@ -0,0 +1,79 @@
+import {useCallback} from 'react';
+
+import {t} from 'sentry/locale';
+import PreferencesStore from 'sentry/stores/preferencesStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import useUrlParams from 'sentry/utils/replays/hooks/useUrlParams';
+import {getDefaultLayout} from 'sentry/views/replays/detail/layout/utils';
+
+export const layoutLabels = {
+  /**
+   * ### Sidebar Right
+   * ┌───────────────────┐
+   * │ Timeline          │
+   * ├──────────┬────────┤
+   * │ Details  > Video  │
+   * │          >        │
+   * │          >^^^^^^^^┤
+   * │          > Crumbs │
+   * │          >        │
+   * └──────────┴────────┘
+   */
+  sidebar_right: t('Player Right'),
+  /**
+   * ### Sidebar Left
+   * ┌───────────────────┐
+   * │ Timeline          │
+   * ├────────┬──────────┤
+   * │ Video  > Details  │
+   * │        >          │
+   * │^^^^^^^ >          |
+   * │ Crumbs >          │
+   * │        >          │
+   * └────────┴──────────┘
+   */
+  sidebar_left: t('Player Left'),
+  /**
+   * ### Topbar
+   *┌────────────────────┐
+   *│ Timeline           │
+   *├───────────┬────────┤
+   *│ Video     │ Crumbs │
+   *│           │        │
+   *├^^^^^^^^^^^^^^^^^^^^┤
+   *│ Details            │
+   *│                    │
+   *└────────────────────┘
+   */
+  topbar: t('Player Top'),
+};
+
+type LayoutKey = keyof typeof layoutLabels;
+
+export function isLayout(val: string): val is LayoutKey {
+  return val in layoutLabels;
+}
+
+function useActiveReplayTab() {
+  const collapsed = !!useLegacyStore(PreferencesStore).collapsed;
+  const defaultLayout = getDefaultLayout(collapsed);
+
+  const {getParamValue, setParamValue} = useUrlParams('l_page', defaultLayout);
+
+  const paramValue = getParamValue();
+
+  return {
+    getLayout: useCallback(
+      (): LayoutKey =>
+        isLayout(paramValue || '') ? (paramValue as LayoutKey) : defaultLayout,
+      [defaultLayout, paramValue]
+    ),
+    setLayout: useCallback(
+      (value: string) =>
+        isLayout(value) ? setParamValue(value) : setParamValue(defaultLayout),
+      [defaultLayout, setParamValue]
+    ),
+  };
+}
+
+export default useActiveReplayTab;
```

Comment:
I would make this a named export so we're not mixing default + named

Reply:
I updated things to pull the labels out of the hook and put them only in the react component that needs them. So that export is gone.

In place I am making an enum available, callsites use that to keep their members in sync. Still two different types of things being exported, only one file imports them both at the same time.